clang/include/clang/Sema/Sema.h

```diff
@@ -13975,7 +13975,7 @@ class Sema final {
     KernelConstStaticVariable
   };
 
-  bool isKnownGoodSYCLDecl(const Decl *D);
+  bool isDeclAllowedInKernel(const Decl *D);
```

Comment:
Is there any justification for this renaming?
I think the old name was a bit more general.

Reply:
`isKnownGoodSYCLDecl` didn't seem to comply with [LLVM's function naming convention](https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly)

The previous name also didn't seem to be correctly phrased. Is known what a good SYCL decl?
The new name re-purposes the same meaning in a better descriptive manner IMO.

> Function names should be verb phrases (as they represent actions), and command-like function should be imperative. The name should be camel case, and start with a lower case letter (e.g. openFile() or isFoo()).
> We cannot stress enough how important it is to use descriptive names. Pick names that match the semantics and role of the underlying entities, within reason.
>

Reply:
I read `isKnownGoodDecl` as we know this is a good/allowed decl, do not diagnose. Anyway I don't mind either name

Reply:
The new naming makes it unclear that this function in intended for SYCL. Other languages (like OpenCL) have kernels as well. Also, there is not only kernels in SYCL where we emit diagnostics.
How about something like `isDeclAllowedInSYCLDeviceCode` ?

Reply:
I literally had the exact same name originally, but was wondering if it'd be too long a name. I'd be more than happy to go with this name. 

> Also, there is not only kernels in SYCL where we emit diagnostics.
Good point!

Reply:
I think it is fine. I've seen longer names in clang.